lib/Runtime/PlatformAgnostic/Platform/Windows/ArraysUtility.cpp

```diff
@@ -0,0 +1,48 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "RuntimePlatformAgnosticPch.h"
+#include "Common.h"
+#include "ChakraPlatform.h"
+
+namespace PlatformAgnostic
+{
+namespace Arrays
+{
+
+    bool GetLocaleSeparator(char16* szSeparator, uint32* sepOutSize, uint32 sepBufSize)
+    {
+        char16 localeName[LOCALE_NAME_MAX_LENGTH];
+
+        int ret = GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH);
+
+        if (ret == 0)
+        {
+            AssertMsg(FALSE, "GetUserDefaultLocaleName failed");
```

Comment:
If GetUserDefaultLocaleName fails, (and it isn't because you passed in invalid parameters), then the system is in a *really* bad state and you probably want to assert and or fail-fast.